haskell/test/Lorentz/Contracts/Test/Permit.hs

```diff
@@ -347,18 +348,49 @@ permitSpec originate = do
           withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
             withSender testPauser $ do
               hash <- callPermit stablecoinContract testPauserPK testPauserSK 0 Pause
-              lCallEP stablecoinContract (Call @"Set_expiry") $ SetExpiryParam 5 (Just (hash, testPauser))
-              lCallEP stablecoinContract (Call @"Set_expiry") $ SetExpiryParam 3 (Just (hash, testPauser))
+              lCallEP stablecoinContract (Call @"Set_expiry") $ SetExpiryParam testPauser 5 $ Just hash
+              lCallEP stablecoinContract (Call @"Set_expiry") $ SetExpiryParam testPauser 3 $ Just hash
             withSender wallet1 $ do
               rewindTime 4
               lCallEP stablecoinContract (Call @"Pause") () `catchExpectedError` errExpiredPermit
 
+      it "can only be accessed by the issuer of the permit" $
+        integrationalTestExpectation $ do
+          withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+            withSender wallet1 $ do
+              pausePermitHash <- callPermit stablecoinContract testPauserPK testPauserSK 0 Pause
+              lCallEP stablecoinContract
+                (Call @"Set_expiry") (SetExpiryParam testPauser 10 $ Just pausePermitHash)
+                `catchExpectedError` errNotPermitIssuer
+
+      it "can be accessed via a permit" $
+        integrationalTestExpectation $ do
+          withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+            withSender wallet1 $ do
+              pausePermitHash <- callPermit stablecoinContract testPauserPK testPauserSK 0 Pause
+              callPermit stablecoinContract testPauserPK testPauserSK 1 $
+                Set_expiry $ SetExpiryParam testPauser 1000 $ Just pausePermitHash
+              lCallEP stablecoinContract (Call @"Set_expiry") $
+                SetExpiryParam testPauser 1000  $ Just pausePermitHash
+              lCallEP stablecoinContract (Call @"Pause") ()
```

Comment:
This does not necessarily proof that the expiry was successfully set.

I think this should work:
1. Setting the default expiry to 1
    ```
        let defaultExpiry = 1
        let originationParams = defaultOriginationParams { opDefaultExpiry = defaultExpiry }
        withOriginated originate originationParams $ \stablecoinContract -> do
    ```
2. issue a permit for `SetExpiry` and then call `SetExpiry` with expiry=5
3. advance time by 3 seconds
4. consume permit by calling `Pause`

Reply:
Done.